arduino-ide-extension/src/node/config-service-impl.ts

```diff
@@ -291,48 +302,64 @@ export class ConfigServiceImpl
   }
 
   private async updateDaemon(
-    port: number | number,
+    port: number,
     config: DefaultCliConfig
   ): Promise<void> {
-    const client = createArduinoCoreServiceClient({ port });
-    const req = new SettingsMergeRequest();
     const json = JSON.stringify(config, null, 2);
-    req.setJsonData(json);
     this.logger.info(`Updating daemon with 'data': ${json}`);
-    return new Promise<void>((resolve, reject) => {
-      client.settingsMerge(req, (error) => {
-        try {
-          if (error) {
-            reject(error);
-            return;
+
+    const updatableConfig = {
+      locale: config.locale,
+      'directories.user': config.directories.user,
+      'directories.data': config.directories.data,
+      'network.proxy': config.network?.proxy,
+      'board_manager.additional_urls':
+        config.board_manager?.additional_urls || [],
+    };
+
+    for (const [key, value] of Object.entries(updatableConfig)) {
+      const client = createArduinoCoreServiceClient({ port });
+      const req = new SettingsSetValueRequest();
```

Comment:
Is this a new CLI behavior? I see a client is created before setting each value. Can it be moved outside of the loop?

Reply:
Yes CLI config API's were refactored and they now only exposed methods for retrieving/setting single values, instead of the previous "merge" approach. 
I agree the client should be created one time outside the loop. Thanks for the review! 👍